duo_client/admin.py

```diff
@@ -519,6 +530,93 @@ def get_authentication_log(self, api_version=1, **kwargs):
+        maxtime (required) - Unix timestamp in ms; fetch records <= maxtime
+        limit - Number of results to limit to
+        next_offset - Used to grab the next set of results from a previous response
+        sort - Sort order to be applied
+
+        Returns:
+            {
+                "items": [
+                    {
+                        "access_device": {
+                            "browser": "Chrome",
+                            "browser_version": "1.11.3",
+                            "ip": {
+                                "address": "1.2.3.4"
+                            },
+                            "location": {
+                                "city": "Ann Arbor",
+                                "country": "United States",
+                                "state": "Michigan"
+                            },
+                            "os": "Mac OS X",
+                            "os_version": "10.14.1"
+                        },
+                        "action": "phone_create",
+                        "activity_id": "edf96844-6e03-4528-963f-1fab8d5fabaa",
+                        "actor": {
+                            "details": "{\"created\": \"2020-01-13T18:56:20.351346+00:00\", \"last_login\": \"2020-02-10T18:56:20.351346+00:00\", \"status\": \"Locked Out\"}",
+                            "key" : "DUABCDECUSTOMER000001",
+                            "name": "John Doe",
+                            "type": "user"
+                        },
+                        "akey": "DABCDECUSTOMER000000",
+                        "application": {
+                            "key": "DIY231J8BR23QK4UKBY8",
+                            "name": "AdminAPI",
+                            "type": "adminapi"
+                        },
+                        "target": {
+                            "details": "{\"number\": \"+19876543210\", \"extension\": \"\"}",
+                            "key" : "DPFZRS9FB0D46QFTM899",
+                            "name": "987-654-3210",
+                            "type": "phone"
+                        },
+                        "ts": "2020-02-13T18:56:20.351346+00:00"
+                    },
+                ],
+                "metadata": {
+                    "next_offset": [
+                        "1532951895000",
+                        "af0ba235-0b33-23c8-bc23-a31aa0231de8"
+                    ],
+                    "total_objects": 1
+                }
+            },
+
+        Raises RuntimeError on error.
+        """
+        params = {}
+
+        for k in kwargs:
+            if kwargs[k] is not None and k in VALID_ACTIVITY_REQUEST_PARAMS:
+                params[k] = kwargs[k]
+
+        if 'mintime' in params:
+            params['mintime'] = str(int(params['mintime']))
+        if 'maxtime' in params:
+            params['maxtime'] = str(int(params['maxtime']))
+        if 'limit' not in params:
```

Comment:
Applying the defaults in the client is unusual for this class; we usually just don't send the parameter, and the Duo cloud side applies the default.  Do you have a specific reason to apply the default here?

For example, get_authentication_log also has a limit and sort parameter, but if those are not provided, they are simply not included in the call (around line 500 above)